holopy/core/io/io.py

```diff
@@ -283,8 +293,29 @@ def load_average(filepath, refimg=None, wavelen=None, index=None, polarization=N
     if len(filepath) < 1:
         raise LoadError(filepath, "No images found")
 
-    accumulator = load_image(filepath[0], refimg.spacing, wavelen, index, polarization)
+    if hasattr(refimg,'spacing') and is_none(spacing):
+        spacing = refimg.spacing
+    
+    accumulator = load_image(filepath[0],spacing)
     for image in filepath[1:]:
-        accumulator += load_image(image)
+        accumulator += load_image(image, spacing)
+    accumulator /= len(filepath)
+
+    if not is_none(refimg):
+        accumulator = update_metadata(accumulator, refimg.medium_index, refimg.illum_wavelen, refimg.illum_polarization, refimg.normals)    
+    return update_metadata(accumulator, medium_index, illum_wavelen, illum_polarization, normals)
+
+def bg_correct(raw, bg, df=None):
+
+    if is_none(df):
+        df = ImageSchema(raw.shape,get_spacing(raw))
+
+    if not (raw.shape == bg.shape == df.shape and list(get_spacing(raw)) == list(get_spacing(bg)) == list(get_spacing(df))):
+        raise BadImage("raw and background images must have the same shape and spacing")
 
-    return accumulator/len(filepath)
+    raw = deepcopy(raw)
+    bg = deepcopy(bg)
+    raw -= df
+    bg -= df
+    raw /= bg
+    return raw
```

Comment:
Consider switching to using regular arithmatic internally and then returning copy_metadata(raw, result)